src/components/patterns/main-map-wrapper/components/MainMapWrapperSubcategory.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+    <div
+        class="vs-main-map-subcategory"
+        data-test="vs-main-map-subcategory"
+    >
+        <BFormGroup
+            label="Using options array:"
+            v-slot="{ ariaDescribedby }"
+        >
+            <BFormCheckboxGroup
+                id="checkbox-group-1"
+                v-model="selected"
+                :aria-describedby="ariaDescribedby"
+                :name="data.name"
+            >
+                <BFormCheckbox
+                    v-for="item in data"
+                    :key="item.id"
+                    :value="item.id"
+                >
+                    {{ item.name }}
+                </BFormCheckbox>
+            </BFormCheckboxGroup>
+        </BFormGroup>
+
+        <div class="vs-main-map-subcategory__controls">
+            <VsButton
+                class="vs-main-map-subcategory__clear-selection"
+                data-test="vs-main-map-subcategory__clear-selection"
+                size="sm"
+                variant="secondary"
+                @click.native="clearSelection"
+            >
+                {{ clearSelectionText }}
+            </VsButton>
+            <VsButton
+                class="vs-main-map-subcategory__apply-filters"
+                data-test="vs-main-map-subcategory__apply-filters"
+                size="sm"
+                @click.native="checkboxesChangeSubmit"
+                :aria-disabled="isDisabled"
+                :disabled="isDisabled"
+            >
+                {{ applyFiltersText }}
+            </VsButton>
+        </div>
+    </div>
+</template>
+
+<script>
+import {
+    BFormGroup,
+    BFormCheckboxGroup,
+    BFormCheckbox,
+} from 'bootstrap-vue';
+import VsButton from '@/components/elements/button/Button';
+
+export default {
+    name: 'VsMainMapWrapperSubCategories',
+    status: 'prototype',
+    release: '0.0.1',
+    components: {
+        BFormGroup,
+        BFormCheckboxGroup,
+        BFormCheckbox,
+        VsButton,
+    },
+    inject: [
+        'clearSelectionText',
+        'applyFiltersText',
+    ],
+    props: {
+        /** Data for subcategory */
+        data: {
+            type: Array,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            selected: [],
+        };
+    },
+    computed: {
+        isDisabled() {
+            if (this.selected.length > 0) {
+                return false;
+            }
+
+            return true;
+        },
+    },
+    methods: {
+        checkboxesChangeSubmit() {
```

Comment:
Can we add some comments to these methods?

Reply:
Added